### DISCUSSION TITLE
Native Maturity Level

### DISCUSSION THREAD

**Discussion by uroybd** [oleeskild/obsidian-digital-garden#187] [Ideas]
We can add an optional maturity level visualization. When I implemented it in my garden, thought it was a very personal choice, not worthy of addition to the upstream. However, on Reddit, I noticed that many are interested in this, and most of them are non-technical.

If we implement it, we have to make the implementation abstracted away.

### Site-wide settings
Toggles for:
1. Show maturity icons on the title
2. Show maturity icons in the file tree
3. Show maturity icons for internal link

### Data
The maturity data will be extracted from the frontmatter by the `maturity` key.

### Defaults
By default, we can provide 3 levels of maturity, but it will be easily simply by CSS variables.

**Comment by uroybd** [comment]
In the current change, I've been able to add all these maturities with just `data-maturity` keys in tags and togglable by classes in the body.

**Comment by oleeskild** [comment]
I think this would be great to support natively. Although it is a sort of "special use case", I think it's very relevant for this plugin as it aims to not only be a generic publishing solution, but a "Digital Garden". I think most digital garden "definitions" have the concept of maturity levels as a big part of it. Allowing users to easily add this information to the notes would be helpful for a lot of people I think.

I think the suggestions you've made on how to implement it is the way to go. We could also add a selector similar to the favicon selector, so that people easily can change their preferred maturity level SVGs from their vault in the plugin settings.

**Reply by uroybd** [reply]
Though I've not added a selector (too much moving parts I think), I made it easily extendable. To change/add any maturity level, one have to add the icon in the img and add one css entry.
